airbyte-webapp/src/hooks/services/useBuildUpdateCheck.ts

```diff
@@ -0,0 +1,55 @@
+import { useEffect } from "react";
+import { useIntl } from "react-intl";
+import { fromEvent, interval, merge, throttleTime } from "rxjs";
+
+import { useAppMonitoringService } from "./AppMonitoringService";
+import { useNotificationService } from "./Notification";
+
+interface BuildInfo {
+  build: string;
+}
+
+const INTERVAL = 60_000;
+
+/**
+ * This hook sets up a check to /buildInfo.json, which is generated by the build system on every build
+ * with a new hash. If it ever detects a new hash in it, we know that the Airbyte instance got updated
+ * and show a notification to the user to reload the page.
+ */
+export const useBuildUpdateCheck = () => {
+  const { formatMessage } = useIntl();
+  const { registerNotification } = useNotificationService();
+  const { trackError } = useAppMonitoringService();
+
+  useEffect(() => {
+    // Trigger the check every ${INTERVAL} milliseconds or whenever the window regains focus again
+    const subscription = merge(interval(INTERVAL), fromEvent(window, "focus"))
+      // Throttle it to maximum once every 10 seconds
+      .pipe(throttleTime(10 * 1000))
+      .subscribe(async () => {
+        try {
+          // Fetch the buildInfo.json file without using any browser cache
+          const buildInfoResp = await fetch("/buildInfo.json", { cache: "no-store" });
```

Comment:
nit: if you parse the response as json in a `then`, there's no need to bind any intermediate variable for the raw response. I'm not big on mixing async/await and then syntaxes randomly, but here I think it makes the intent clearer by reducing "fetch some json from a file" to a single step.
```ts
const buildInfo: BuildInfo = await fetch("/buildInfo.json", { cache: "no-store" }).then((res) => res.json());
```